modules/javafx.controls/src/main/java/com/sun/javafx/scene/control/LambdaMultiplePropertyChangeListenerHandler.java

```diff
@@ -25,40 +25,96 @@
 
 package com.sun.javafx.scene.control;
 
-import javafx.beans.value.ChangeListener;
-import javafx.beans.value.ObservableValue;
-import javafx.beans.value.WeakChangeListener;
-
 import java.util.HashMap;
+import java.util.IdentityHashMap;
 import java.util.Map;
 import java.util.function.Consumer;
 
+import javafx.beans.InvalidationListener;
+import javafx.beans.Observable;
+import javafx.beans.WeakInvalidationListener;
+import javafx.beans.value.ChangeListener;
+import javafx.beans.value.ObservableValue;
+import javafx.beans.value.WeakChangeListener;
+import javafx.collections.ListChangeListener;
+import javafx.collections.ListChangeListener.Change;
+import javafx.collections.ObservableList;
+import javafx.collections.WeakListChangeListener;
+
+/**
+ * Handler to manage multiple listeners to multiple observables. It handles
+ * adding/removing listeners for common notification events (change,
+ * invalidation and list change) on particular instances of observables. The
+ * listeners are wrapped into their weak counterparts to minimize the potential of
+ * memory leaks.
+ * <p>
+ * Clients can register consumers to be invoked on receiving notification.
+ * Un-/Registration api is separate per notification type and per observable.
+ * It is allowed to register multiple consumers per observable. They
+ * are executed in the order of registration. Note that unregistration
+ * of a given observable stops observing that observable (for the notification
+ * type of the unregistration) completely, that is none of the consumers
+ * previously registered with this handler will be executed after unregistering.
+ * <p>
+ * Disposing removes all listeners added by this handler from all registered observables.
+ *
+ */
 public final class LambdaMultiplePropertyChangeListenerHandler {
+// FIXME: name doesn't fit after widening to support more notification event types
```

Comment:
Maybe `MultipleObservableListenersHandler`? I didn't look at the class too much.

Reply:
I like `MultipleObservableListenersHandler` or `LambdaMultipleObservableListenersHandler` if you want to keep `Lambda` in the name (not sure it matters much).

Reply:
sorry, somehow this comment didn't reach my consciousness ;) `MultipleObservableLIstenersHandler` sounds fine. Would like to extract the rename to a follow-up issue, though, because it's not only used by SkinBase.